AWSAppSyncClient/Internal/NetworkReachability.swift

```diff
@@ -8,14 +8,14 @@ import Foundation
 import Reachability
 
 /// Defines a factory to return a NetworkReachabilityProviding instance
-protocol NetworkReachabilityProvidingFactory {
+public protocol NetworkReachabilityProvidingFactory {
```

Comment:
Please move the newly-public protocols and extensions out of the `Internal` folder into a top-level file called `NetworkReachability.swift`, and let's rename this file to `Internal/NetworkReachabilityNotifier.swift`